conda_build/environ.py

```diff
@@ -47,6 +47,10 @@ def get_git_build_info(src_dir, git_url, expected_rev):
     env = os.environ.copy()
     d = {}
     git_dir = join(src_dir, '.git')
+    if not isinstance(git_dir, str):
```

Comment:
Could we just do something `env['GIT_DIR'] = str(git_dir)`  In Python 2 it would cast unicode -> str.  In Python 3, it wouldn't do anything, since it's already a string.